### PR TITLE
Work manager

### PR DESCRIPTION
**Pull request by tregubowww**

### CODE REVIEW

File: app/src/main/java/ru/myuniquenickname/myapplication/data/work_manager/WorkRepository.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package ru.myuniquenickname.myapplication.data.work_manager
+
+import androidx.work.Constraints
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.PeriodicWorkRequest
+import java.util.concurrent.TimeUnit
+
+class WorkRepository {
```
Comment (pcfaktor) [inline app/src/main/java/ru/myuniquenickname/myapplication/data/work_manager/WorkRepository.kt]: I think the name of the Repository pattern is not fit here.

File: app/src/main/java/ru/myuniquenickname/myapplication/presentation/movie_list/MovieListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentMoviesListBinding.inflate(inflater, container, false)
+        adapterMovies = MoviesAdapter(clickListener)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        WorkManager.getInstance(requireContext()).enqueue(workRepository.constrainedRequest)
```
Comment (pcfaktor) [inline app/src/main/java/ru/myuniquenickname/myapplication/presentation/movie_list/MovieListFragment.kt]: Why isn't it the responsibility of the ViewModel?

**Review by pcfaktor — APPROVED** [review]
LGTM

File: app/src/main/java/ru/myuniquenickname/myapplication/presentation/movie_list/MovieListFragment.kt (hunk 2 above)
Comment (tregubowww) [reply app/src/main/java/ru/myuniquenickname/myapplication/presentation/movie_list/MovieListFragment.kt]: I need a context to create, isn't it better to create here?

**Pull request merged**